.azure-pipelines/main-pipeline.yml

```diff
@@ -116,15 +116,27 @@ stages:
               vsixFile: '$(Pipeline.Workspace)/$(artifactPath)/$(packagedTask)'
               extensionTag: '$(devExtensionTag)'
               extensionName: '$(extensionName) DEV'
-              extensionVersion: '$(queryDevVersion.Extension.Version)'
-              updateTasksId: true
               extensionVisibility: 'private'
               extensionPricing: 'free'
               shareWith: 'datadog-ci'
+              extensionVersion: '$(queryDevVersion.Extension.Version)'
+              updateTasksId: true # The DEV task ID is overridden to 99cf84cd-0117-58b7-b1dd-d95e5ae9baf8 (based on the `extensionTag`)
+              updateTasksVersion: true # Do not require updating the task version during development
```

Comment:
This is **required**.

> You must rev **both the task version and extension version** for an update to occur. [Source](https://learn.microsoft.com/en-us/azure/devops/extend/develop/add-build-task?view=azure-devops#4-package-your-extension)

We do not set `updateTasksVersion: true` below for the public extension, because bumping the task version is done manually ([example PR](https://github.com/DataDog/datadog-ci-azure-devops/pull/58/files)) - and we'll later do it automatically with our Release Automation system.

Reply:
You can find documentation about all the inputs here: https://github.com/microsoft/azure-devops-extension-tasks/blob/main/BuildTasks/PublishExtension/v4/task.json